src/game/server/swarm/asw_marine.cpp

```diff
@@ -1624,8 +1624,8 @@ int CASW_Marine::OnTakeDamage_Alive( const CTakeDamageInfo &info )
 		}
 	}
 
-	// scale down the damage received by bots but not for PvP
-	if ( rd_bot_strong.GetBool() && !IsInhabited() && !ASWDeathmatchMode() )
+	// scale down the damage received by bots but not for PvP, jh: make marines take standard damage from infestation/burning if marine was inhabited when got infested/ignited (prevent people exploiting damage taken by going bot)
+	if ( rd_bot_strong.GetBool() && !IsInhabited() && !ASWDeathmatchMode() && !( IsInfested() && m_bGotInfestedWhenInhabited && info.GetDamageType() & DMG_INFEST ) && !( IsOnFire() && m_bGotIgnitedWhenInhabited && info.GetDamageType() & DMG_BURN ) )
```

Comment:
```suggestion
	if ( rd_bot_strong.GetBool() && !IsInhabited() && !ASWDeathmatchMode() &&
		!( IsInfested() && m_bGotInfestedWhenInhabited && ( info.GetDamageType() & DMG_INFEST ) ) &&
		!( IsOnFire() && m_bGotIgnitedWhenInhabited && ( info.GetDamageType() & ( DMG_BURN | DMG_DIRECT ) ) == ( DMG_BURN | DMG_DIRECT) ) )
```
Split into multiple lines and made burning check only affect damage-over-time burning damage.